x-pack/plugins/monitoring/server/config.ts

```diff
@@ -4,92 +4,24 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { schema, TypeOf } from '@kbn/config-schema';
+import {
+  configSchema as elasticsearchConfigSchema,
+  ElasticsearchConfig,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../../src/core/server/elasticsearch';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { ElasticsearchConfigType } from '../../../../src/core/server/elasticsearch/elasticsearch_config';
 
 const hostURISchema = schema.uri({ scheme: ['http', 'https'] });
-const DEFAULT_API_VERSION = 'master';
+
+const monitoringElasticsearchConfigSchema = elasticsearchConfigSchema.extends({
+  logFetchCount: schema.number({ defaultValue: 10 }),
+  hosts: schema.maybe(schema.oneOf([hostURISchema, schema.arrayOf(hostURISchema, { minSize: 1 })])),
```

Comment:
Do we really need to override the `hosts`? Since, it's already defined [here](https://github.com/chrisronline/kibana/blob/321f3eed6d3778e65a4f231217fcb924b7f4055a/src/core/server/elasticsearch/elasticsearch_config.ts#L45). Or, are you trying to avoid the default value?

Reply:
@igoristic Exactly, the monitoring elasticsearch config can have an undefined hosts and that means we use the `elasticsearch.*` settings instead.